.github/workflows/build-and-deploy-images.yml

```diff
@@ -40,25 +42,32 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
-      - name: Establish build version environment variable
-        run: echo "BUILD_VERSION=$(cat version_number.txt).$GITHUB_RUN_NUMBER" >> "$GITHUB_ENV"
+      - name: Set BUILD_VERSION for Staging
+        if: github.ref == 'refs/heads/staging'
+        run: echo "BUILD_VERSION=staging-$GITHUB_SHA" >> "$GITHUB_ENV"
+        run: echo "BUILD_VERSION_LATEST=staging" >> "$GITHUB_ENV"
+
+      - name: Set BUILD_VERSION for Production from git tag
+        if: contains(github.ref, 'refs/tags/')
+        # define GIT_TAG as stripping off initial 11 characters, i.e. "refs/tags/v"
+        run: echo "GIT_TAG=$(echo ${GITHUB_REF:11})" >> $GITHUB_ENV
+        run: echo "BUILD_VERSION=$GITHUB_TAG" >> "$GITHUB_ENV"
```

Comment:
I'm pretty sure you meant `GIT_TAG` here. As far as I can tell, `GITHUB_TAG` isn't an env var that GitHub Actions provides, so this would make `BUILD_VERSION` an empty string.

Reply:
done.